richtext-ui/src/commonMain/kotlin/com/halilibo/richtext/ui/string/Text.kt

```diff
@@ -68,6 +154,77 @@ public fun RichTextScope.Text(
   }
 }
 
+private data class TextAnimation(val startIndex: Int, val alpha: Float)
+
+private fun AnnotatedString.animateAlphas(animations: Collection<TextAnimation>, contentColor: Color): AnnotatedString {
+  if (this.text.isEmpty() || animations.isEmpty()) {
+    return this
+  }
+  var remainingText = this
+  val modifiedTextSnippets = mutableStateListOf<AnnotatedString>()
+  animations.sortedByDescending { it.startIndex }.forEach { animation ->
+    if (animation.startIndex < remainingText.length) {
+      modifiedTextSnippets.add(
+        remainingText.subSequence(animation.startIndex, remainingText.length)
+          .changeAlpha(animation.alpha, contentColor)
+      )
+      remainingText = remainingText.subSequence(0, animation.startIndex)
+    } else {
+    }
+  }
+  return AnnotatedString.Builder(remainingText).apply {
+    modifiedTextSnippets.reversed().forEach { append(it) }
+  }.toAnnotatedString()
+//  return modifiedTextSnippets.reversed().fold(remainingText) { acc, snippet -> acc + snippet }
+}
+
+private fun AnnotatedString.changeAlpha(alpha: Float, contentColor: Color): AnnotatedString {
+  val newWordsStyles = spanStyles.map { spanstyle ->
+    spanstyle.copy(item = spanstyle.item.copy(color = spanstyle.item.color.copy(alpha = alpha)))
+  }.ifEmpty {
+    listOf(AnnotatedString.Range(SpanStyle(contentColor.copy(alpha = alpha)), 0, length))
+  }
+  return AnnotatedString(text, newWordsStyles)
+}
+
+// Given a string, finds all the "phrases" (sentences, commas, and any word sequences longer than 5 words)
+// and returns a list of all their starting indexes
```

Comment:
This style is required for it to show up in the IDE documentation popup
```suggestion
/**
  *  Given a string, finds all the "phrases" (sentences, commas, and any word sequences longer than 5 words)
  * and returns a list of all their starting indexes
  */
```